OneSchedule.Domain/NotificationScheduler.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Microsoft.Extensions.Configuration;
+using OneSchedule.Domain.Abstractions;
+using OneSchedule.Entities;
+using Quartz;
+using Quartz.Impl;
+using Quartz.Lambda;
+using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Telegram.Bot;
+
+namespace OneSchedule.Domain
+{
+    public class NotificationScheduler : INotificationScheduler
+    {
+        private readonly ITelegramBotClient _bot;
+        private readonly IConfiguration _configuration;
+
+        public NotificationScheduler(ITelegramBotClient bot, IConfiguration configuration)
+        {
+            _bot = bot;
+            _configuration = configuration;
+        }
+
+        public async Task ScheduleNotifications(EventEntity eventEntity)
+        {
+            IScheduler scheduler = await SchedulerRepository.Instance.Lookup(_configuration.GetSection("Quartz").GetSection("quartz.scheduler.instanceName").Value);
+
+            List<Task> tasks = new List<Task>(new Task[eventEntity.Notifications.Count]);
```

Comment:
I don't know how you did that, but as far as I know constructor accepts the length of the list